app/models/update_log.rb

```diff
@@ -1,17 +1,62 @@
 class UpdateLog
+  ####################
+  # CONSTANTS        #
+  ####################
+  MAX_UPDATES_TO_KEEP = 10
 
   def self.setting_record
     Setting.find_or_create_by(key: 'metrics_update')
   end
 
-  def self.log_update(time)
+  def self.log_updates(starttime, endtime)
+    self.add_new_log
+    self.update_data(starttime, endtime)
+    self.log_delay
+  end
+
+  def self.average_delay
     setting = self.setting_record
-    setting.value["last_update"] = time
-    setting.save
+    setting.value['average_delay'] ? setting.value['average_delay']: nil
   end
 
   def self.last_update
-    self.setting_record
-    Setting.find_by(key: 'metrics_update').value['last_update']
+    setting = self.setting_record
+    setting.value['constant_update'] ? setting.value['constant_update'].values.last['end_time'] : nil
   end
+
+  private 
+
+    def self.add_new_log
```

Comment:
This looks like it just initializes the hash if it's not already there. That doesn't match up with the method name. My suggestion is to have `add_new_log` insert the start/end time from the new update (and first initialize the hash, if necessary), then have a separate method to remove oldest one if there are more than 10 updates already stored.